geomesa-blobstore/geomesa-blobstore-core/src/main/scala/org/locationtech/geomesa/blob/core/AccumuloBlobStore.scala

```diff
@@ -89,83 +85,84 @@ class AccumuloBlobStore(ds: AccumuloDataStore) extends LazyLogging with BlobStor
     }
   }
 
-  def delete(id: String): Unit = {
-    // TODO: Get Authorizations using AuthorizationsProvider interface
-    // https://geomesa.atlassian.net/browse/GEOMESA-986
-    val bd = connector.createBatchDeleter(blobTableName, new Authorizations(), bwc.getMaxWriteThreads, bwc)
-    bd.setRanges(List(new Range(new Text(id))))
-    bd.delete()
-    bd.close()
-    deleteFeature(id)
-  }
-
-  private def deleteFeature(id: String): Unit = {
-    val removalFilter = Filters.ff.id(Filters.ff.featureId(id))
-    val fd = ds.getFeatureWriter(blobFeatureTypeName, removalFilter, Transaction.AUTO_COMMIT)
+  private def getInternal(id: String): (String, Array[Byte]) = {
+    val scanner = connector.createScanner(
+      blobTableName,
+      ds.authProvider.getAuthorizations
+    )
     try {
-      while (fd.hasNext) {
-        fd.next()
-        fd.remove()
+      scanner.setRange(new Range(new Text(id)))
+
+      val iter = SelfClosingIterator(scanner)
+      if (iter.hasNext) {
+        val next = iter.next()
+        val ret = (next.getKey.getColumnQualifier.toString, next.getValue.get)
+        iter.close()
```

Comment:
you've got some duplicate closing going on here, plus the selfClosing, plus the finally